### PR TITLE
Add upstream doc for PriorityClass

### PR DESCRIPTION
**Pull request by Vincent056**
Add upstream doc for `PriorityClass`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,8 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 - Added a template for proposing and discussing
   [enhancements](https://github.com/ComplianceAsCode/compliance-operator/tree/master/enhancements).
 
+- Added upstream documentation for `PriorityClass`
```
Comment (rhmdnd) [inline CHANGELOG.md:23]: We can probably omit this since we're documenting it in the enhancements section in https://github.com/ComplianceAsCode/compliance-operator/pull/71/files

File: doc/usage.md [hunk 2]
```diff
@@ -320,3 +320,61 @@ oc run --rm -i --restart=Never --image=registry.fedoraproject.org/fedora-minimal
 rer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" https://metrics.openshift-compliance.svc:8585/metrics-co' | grep compliance
 ```
 
+## To use PriorityClass for scans
+
+When heavily using Pod Priority and Preemption for automated scaling and
+the default PriorityClass is too low to guarantee pods to run then scans
+are not executed and reports are missing. Since the Compliance Operator
+is important for ensuring compliance, we should give administrators the
+ability to associate a PriorityClass to the operator. This will ensure
```
Comment (rhmdnd) [inline doc/usage.md:329]: > Since the Compliance Operator
is important for ensuring compliance, we should give administrators the
ability to associate a PriorityClass to the operator

I would reword this to be less opinionated about importance (example):

Clusters that must have compliance evidence can use pod priority with a `ScanSetting` to ensure scans are executed.

File: doc/usage.md [hunk 3]
```diff
@@ -320,3 +320,61 @@ oc run --rm -i --restart=Never --image=registry.fedoraproject.org/fedora-minimal
...
+roles:
+  - master
+  - worker
+scanTolerations:
+  - operator: Exists
+```
+
+If the `PriorityClass` referenced in the `ScanSetting` can't be found,
+the operator will leave `PriorityClass` empty and issue a warning in 
+the event.
```
Comment (rhmdnd) [inline doc/usage.md:378]: We could also say that the compliance operator will continue to schedule scans (invalid Priority Classes won't fail a scan).

File: doc/usage.md [hunk 4]
```diff
@@ -320,3 +320,61 @@ oc run --rm -i --restart=Never --image=registry.fedoraproject.org/fedora-minimal
 rer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" https://metrics.openshift-compliance.svc:8585/metrics-co' | grep compliance
 ```
 
+## To use PriorityClass for scans
+
+When heavily using Pod Priority and Preemption for automated scaling and
```
Comment (rhmdnd) [inline doc/usage.md:325]: We could link Pod Priority and Preemption to the documentation you linked below.

https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/

File: doc/usage.md [hunk 5]
```diff
@@ -320,3 +320,61 @@ oc run --rm -i --restart=Never --image=registry.fedoraproject.org/fedora-minimal
...
+  - worker
+scanTolerations:
+  - operator: Exists
+```
+
+If the `PriorityClass` referenced in the `ScanSetting` can't be found,
+the operator will leave `PriorityClass` empty and issue a warning in 
+the event.
+
+[1][Kubernetes documentation on this](https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/)
```
Comment (rhmdnd) [inline doc/usage.md:380]: We could remove this if we include it above.

File: doc/usage.md [hunk 6]
```diff
@@ -320,3 +320,62 @@ oc run --rm -i --restart=Never --image=registry.fedoraproject.org/fedora-minimal
...
+  - master
+  - worker
+scanTolerations:
+  - operator: Exists
+```
+
+If the `PriorityClass` referenced in the `ScanSetting` can't be found,
+the operator will leave `PriorityClass` empty and issue a warning in 
+the event, the Compliance Operator will continue to schedule scans
+without a `PriorityClass`.
```
Comment (rhmdnd) [inline doc/usage.md:379]: One minor suggestion.

If the `PriorityClass` referenced in the `ScanSetting` can't be found,
the operator will leave `PriorityClass` empty, issue a warning, and continue scheduling scans
without a `PriorityClass`.

**Review by rhmdnd — COMMENTED** [review]
Looks great, one minor comment on wording.

File: doc/usage.md (hunk 6 above)
Comment (Vincent056) [reply doc/usage.md]: thanks for the review, just updated it

**Review by JAORMX — APPROVED** [review]

**Comment by rhmdnd** [comment]
Tagging @mkumku for PX review.
